### ISSUE TITLE
Form control values do not get set with trailing space, making email validator difficult to diagnose

### ISSUE THREAD

**Issue by BobDankert** [angular/angular#54827]
### Which @angular/* package(s) are the source of the bug?

forms

### Is this a regression?

No

### Description

When calling setValue() on a FormControl with a value that contains a trailing space, the input element value does not receive that trailing space (however the FormControl value will have the trailing space). This can make it difficult on email form controls with the email validator, as a trailing space will fail the email validator, but the input value does not have the trailing space. To "fix" the issue, the user needs to modify the email in the input in anyway (add a space, oddly enough, work) to reset the FormControl value to something that matches the input value.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-bhq16g?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
The input value does not match the FormControl value.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.2.3
Node: 18.18.0
Package Manager: npm 10.2.3
OS: linux x64

Angular: 17.2.4
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1702.3
@angular-devkit/build-angular   17.2.3
@angular-devkit/core            17.2.3
@angular-devkit/schematics      17.2.3
@angular/cli                    17.2.3
@schematics/angular             17.2.3
rxjs                            7.8.1
typescript                      5.3.3
zone.js                         0.14.4
```

### Anything else?

I don't know if it is the browser not setting the trailing space, or something internal to the FormControl setValue() call that does not set the trailing space, but this is certainly creating an odd situation, as shown in the StackBlitz reproduction.

**Comment by jnizet** [comment]
Just a note: this seems to be specific to inputs of type "email". If the type of the input is "text", the space is present in the input value.
And this happens in both Chrome and Fireforx

**Comment by alxhub** [comment]
This is a super weird case, for sure.

**Comment by alexkunin** [comment]
Let me add some context to this, guys.

According to [HTML specs](https://html.spec.whatwg.org/multipage/input.html#email-state-(type=email)) of `type=email` (this is for non-`multiple` mode; `multiple` also has similar behavior, a bit different tho):

> The value sanitization algorithm is as follows: Strip newlines from the value, then strip leading and trailing ASCII whitespace from the value.

When [value sanitization algorithm](https://html.spec.whatwg.org/multipage/input.html#value-sanitization-algorithm) is applied (there are some other cases on that page too, but this one is most relevant):

> When the *value content attribute is added, set, or removed*, if the control's dirty value flag is false, the user agent must set the value of the element to the value of the value content attribute, if there is one, or the empty string otherwise, and *then run the current value sanitization algorithm*, if one is defined.

So, basically, as with `type=number` etc., best way to handle this is to avoid using anything other than `type=text` (with notable `password`, `radio`, `file` etc. exceptions).

Not 100% sure, but maybe specialized `ControlValueAccessor` for `input[type=email][formControlName]` etc. is the correct way to resolve this?

**Comment by JeanMeche** [comment]
I'll close this as a native behavior of the email inputs.

**Comment by JeanMeche** [comment]
Note: In 21.1, signal forms exhibits the same issue.
